Enterprise-CMCS/managed-care-review
# Clean empty rates, fix hasMissingField UI for rate summary page

## Summary
- Adds new data migration to delete empty rates (relevant in particular for Val)
- Fix UI bug where we were showing the "missing field" red error for CMS users. This is a state user feature .

#### Related issues
https://qmacbis.atlassian.net/browse/MCR-3567

#### Screenshots

#### Test cases covered
- Add tests for back link and document download button to `RateSummary.test.tsx` (we handle form field render tests already on `SingleRateSummarySection` tests
- Add tests to `SingleRateSummarySection.test.tsx` for the explainMissingData behavior



## QA guidance
- Pull down val db locally then run my migration - it should delete about 16 rates - all those with the missing fields. 
- For `explainMissingData`- I suggest just  test this this locally by creating a bad submission (for example delete a required field in a submitted rate) and then going to rates dashboard. CMS users should **not** get missing text explainer, they just see an empty field.
